### PR TITLE
[telemetry] Save and read application state on a background thread.

### LINKED ISSUE #537: android.os.strictmode.DiskReadViolation in Strict mode
**Configuration**
 - SDK Version [e.g. 1.9.x] 7.0.3
 - Android OS Version [ e.g. 8.0.0] 10.0
 - Device model(s): [e.g. Google Pixel XL etc] OPPO PBEM00
 - App: [e.g. Mapbox Studio] Maps test app

**Steps to Reproduce**
How did you get there?
Enable the strict mode in the [test app ](https://github.com/mapbox/mapbox-maps-android/blob/main/app/src/main/java/com/mapbox/maps/testapp/MapboxApplication.kt#L17)with the following codes:
```
 StrictMode.setThreadPolicy(
      StrictMode.ThreadPolicy.Builder()
        .detectAll()
        .penaltyLog()
        .penaltyDeath()
        .build()
    )
```
Then run the test app and it crashes.
**Expected**
What did you expect?
The app runs normally without crash
**Logging**
```
  java.lang.RuntimeException: StrictMode ThreadPolicy violation
        at android.os.StrictMode$AndroidBlockGuardPolicy.onThreadPolicyViolation(StrictMode.java:1768)
        at android.os.StrictMode$AndroidBlockGuardPolicy.lambda$handleViolationWithTimingAttempt$0$StrictMode$AndroidBlockGuardPolicy(StrictMode.java:1687)
        at android.os.-$$Lambda$StrictMode$AndroidBlockGuardPolicy$9nBulCQKaMajrWr41SB7f7YRT1I.run(Unknown Source:6)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:238)
        at android.app.ActivityThread.main(ActivityThread.java:7798)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:995)
     Caused by: android.os.strictmode.DiskReadViolation
        at android.os.StrictMode$AndroidBlockGuardPolicy.onReadFromDisk(StrictMode.java:1556)
        at libcore.io.BlockGuardOs.access(BlockGuardOs.java:69)
        at libcore.io.ForwardingOs.access(ForwardingOs.java:73)
        at android.app.ActivityThread$AndroidOs.access(ActivityThread.java:7688)
        at java.io.UnixFileSystem.checkAccess(UnixFileSystem.java:281)
        at java.io.File.exists(File.java:815)
        at android.app.ContextImpl.getDataDir(ContextImpl.java:2440)
        at android.app.ContextImpl.getPreferencesDir(ContextImpl.java:603)
        at android.app.ContextImpl.getSharedPreferencesPath(ContextImpl.java:800)
        at android.app.ContextImpl.getSharedPreferences(ContextImpl.java:452)
        at android.content.ContextWrapper.getSharedPreferences(ContextWrapper.java:189)
        at com.mapbox.android.telemetry.AppStateUtils.saveActivityState(AppStateUtils.java:84)
        at com.mapbox.android.telemetry.MapboxTelemetryService.resetActivityStateToUnknown(MapboxTelemetryService.java:78)
        at com.mapbox.android.telemetry.MapboxTelemetryService.onCreate(MapboxTelemetryService.java:84)
        at android.app.ActivityThread.handleCreateService(ActivityThread.java:4221)
        at android.app.ActivityThread.access$200(ActivityThread.java:240)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2088)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:238) 
        at android.app.ActivityThread.main(ActivityThread.java:7798) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:995) 
```

### PR DESCRIPTION
**Pull request by tarigo**
To avoid delays (~3-5ms) on the main thread.
Fixes  #537.

### CODE REVIEW

File: libtelemetry/src/full/java/com/mapbox/android/telemetry/AppStateUtils.java [hunk 1]
```diff
@@ -196,11 +207,20 @@ private static AppState arbitrage(AppState stateFromActivityManager,
...
-    return state;
+  public interface GetAppStateCallback {
+    void onReady(AppState state);
+  }
+
+  public static void getAppState(@NonNull final Context context, final GetAppStateCallback callback) {
+    ioExecutor.execute(new Runnable() {
+      @Override
+      public void run() {
+        Log.i(TAG, "Getting app state...");
```
Comment (mr1sunshine) [inline libtelemetry/src/full/java/com/mapbox/android/telemetry/AppStateUtils.java:218]: Do we want to keep those logs?

File: libtelemetry/src/full/java/com/mapbox/android/telemetry/location/LocationUpdatesBroadcastReceiver.java [hunk 2]
```diff
@@ -39,15 +39,20 @@ public void onReceive(Context context, Intent intent) {
...
+      final String sessionId = collectionClient.getSessionId();
       List<Location> locations = result.getLocations();
-      for (Location location : locations) {
+      for (final Location location : locations) {
         if (isThereAnyNaN(location) || isThereAnyInfinite(location)) {
           continue;
         }
-        final String appState = AppStateUtils.getAppState(context).toString();
-        telemetry.push(LocationMapper.create(location, appState, sessionId));
+        AppStateUtils.getAppState(context, new AppStateUtils.GetAppStateCallback() {
```
Comment (mr1sunshine) [inline libtelemetry/src/full/java/com/mapbox/android/telemetry/location/LocationUpdatesBroadcastReceiver.java:49]: Do we want to start new thread execution for each location? Would it be better to read state first and apply for the whole location array?

File: libtelemetry/src/full/java/com/mapbox/android/telemetry/AppStateUtils.java (hunk 1 above)
Comment (tarigo) [reply libtelemetry/src/full/java/com/mapbox/android/telemetry/AppStateUtils.java]: Ops, by accident, moving back to `Logger.v`.

File: libtelemetry/src/full/java/com/mapbox/android/telemetry/location/LocationUpdatesBroadcastReceiver.java (hunk 2 above)
Comment (tarigo) [reply libtelemetry/src/full/java/com/mapbox/android/telemetry/location/LocationUpdatesBroadcastReceiver.java]: Defenitely.

**Review by mr1sunshine — APPROVED** [review]
LGTM

**Pull request merged**